Node/src/models/o365ConnectorCard.ts

```diff
@@ -0,0 +1,577 @@
+        if (text) {
+            this.data.text = fmtText(this.session, text, args);
+        }
+        return this;
+    }
+
+    public activityTitle(text: string|string[], ...args: any[]): this {
+        if (text) {
+            this.data.activityTitle = fmtText(this.session, text, args);
+        }
+        return this;
+    }
+
+    public activitySubtitle(text: string|string[], ...args: any[]): this {
+        if (text) {
+            this.data.activitySubtitle = fmtText(this.session, text, args);
+        }
+        return this;
+    }
+
+    public activityText(text: string|string[], ...args: any[]): this {
+        if (text) {
+            this.data.activityText = fmtText(this.session, text, args);
+        }
+        return this;
+    }
+
+    public activityImage(imageUrl: string): this {
+        if (imageUrl) {
+            this.data.activityImage = imageUrl;
+        }
+        return this;
+    }
+
+    public markdown(flag: boolean): this {
+        this.data.markdown = !!flag;
+        return this;
+    }
+
+    public facts(list: teams.IO365ConnectorCardFact[]|teams.IIsO365ConnectorCardFact[]): this {
+        this.data.facts = [];
+        if (list) {
+            for (var i = 0; i < list.length; i++) {
+                var fact = list[i];
+                this.data.facts.push((<teams.IIsO365ConnectorCardFact>fact).toFact ? (<teams.IIsO365ConnectorCardFact>fact).toFact() : <teams.IO365ConnectorCardFact>fact);    
+            }
+        }
+        return this;
+    }
+
+    public images(list: teams.IO365ConnectorCardImage[]|teams.IIsO365ConnectorCardImage[]): this {
+        this.data.images = [];
+        if (list) {
+            for (var i = 0; i < list.length; i++) {
+                var image = list[i];
+                this.data.images.push((<teams.IIsO365ConnectorCardImage>image).toImage ? (<teams.IIsO365ConnectorCardImage>image).toImage() : <teams.IO365ConnectorCardImage>image);
+            }
+        }
+        return this;
+    }
+    
+    public potentialAction(list: teams.IO365ConnectorCardActionBase[]|teams.IIsO365ConnectorCardActionBase[]): this {
+        this.data.potentialAction = [];
+        if (list) {
+            for (var i = 0; i < list.length; i++) {
+                var action = list[i];
+                var obj = (<teams.IIsO365ConnectorCardActionBase>action).toAction ?
+                          (<teams.IIsO365ConnectorCardActionBase>action).toAction() : <teams.IO365ConnectorCardActionBase>action;
+                this.data.potentialAction.push(o365ActionToPayload(obj));
+            }
+        }
+        return this;
+    }
+
+    public toSection(): teams.IO365ConnectorCardSection {
+        return this.data;
+    }
+}
+
+export class O365ConnectorCardFact implements teams.IIsO365ConnectorCardFact {
+    private data = <teams.IO365ConnectorCardFact>{ name: '' };
+    
+    constructor(private session?: builder.Session) {
+        
+    }
+    
+    public name(v: string): this {
```

Comment:
I copied the implementation for CardFact from BotBuilder. Apparently for some reasons they didn't do so, but anyway in update 2 I revised it and made homogeneous implementations for both name & value.